test/cri-containerd/helpers/log.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/Microsoft/go-winio"
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	soutPipe := os.Getenv("CONTAINER_STDOUT")
+	waitPipe := os.Getenv("CONTAINER_WAIT")
+
+	sout, err := winio.DialPipeContext(ctx, soutPipe)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "couldn't open pipe. %s", err)
+		os.Exit(1)
+	}
+	defer sout.Close()
+
+	wait, err := winio.DialPipeContext(ctx, waitPipe)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "couldn't open pipe. %s", err)
+		os.Exit(1)
+	}
+
+	destPath := os.Args[1]
+	dest, err := os.Create(destPath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "couldn't open destination file: %s", err)
```

Comment:
Are we supposed to be `os.Exit(1)`'ing here also?

Reply:
If so, we need to defer `wait.Close()` above in case this errors out. It should be safe to call multiple times.